packages/jest-runtime/src/index.ts

```diff
@@ -718,12 +718,16 @@ class Runtime {
     try {
       fn();
     } finally {
+      this._isolatedModuleRegistry.clear();
+      this._isolatedMockRegistry.clear();
       this._isolatedModuleRegistry = null;
       this._isolatedMockRegistry = null;
     }
   }
 
   resetModules(): void {
+    this._isolatedModuleRegistry?.clear();
+    this._isolatedMockRegistry?.clear();
     this._isolatedModuleRegistry = null;
     this._isolatedMockRegistry = null;
```

Comment:
do we need to set these to null? other registries rely on `clear()` only

Reply:
this is for `jest.isolateModules` which relies on them being `null` to swap implementations